src/btree/bt_split.c

```diff
@@ -1154,17 +1155,17 @@ __split_internal(WT_SESSION_IMPL *session, WT_PAGE *parent, WT_PAGE *page)
         }
         __wt_free_ref_index(session, page, alloc_index, true);
         break;
-    case WT_ERR_PANIC:
-        __wt_err(session, ret, "fatal error during internal page split");
-        ret = WT_PANIC;
-        break;
     case WT_ERR_IGNORE:
-        if (ret != 0 && ret != WT_PANIC) {
-            __wt_err(session, ret,
-              "ignoring not-fatal error during internal page "
-              "split");
+        if (ret == 0)
+            break;
+        if (ret != WT_PANIC) {
+            __wt_err(session, ret, "ignoring not-fatal error during internal page split");
             ret = 0;
+            break;
```

Comment:
The way this turned out is kind of unsatisfying.

Reply:
I'm totally open to something better, what would you suggest?

Reply:
The term "better" is clearly subjective. What about something like:
```
if (ret != WT_PANIC) {
    if (ret != 0)
        __wt_err(...)
    ret = 0;
    break;
}
```

Reply:
OK, agreed.

Pushed in 3d9cdc280.